src/impressions/counter.js

```diff
@@ -0,0 +1,30 @@
+const TIME_INTERVAL_MS = 3600*1000;
+
+class ImpressionCounter {
+  constructor() {
+    this.cache = {};
+  }
+
+  _truncateTimeFrame(timestampInMs) {
+    return timestampInMs - (timestampInMs % TIME_INTERVAL_MS);
+  }
+
+  _makeKey(featureName, timeFrame) {
+    return `${featureName}::${this._truncateTimeFrame(timeFrame)}`;
+  }
+
+  inc(featureName, timeFrame, amount) {
+    const key = this._makeKey(featureName, timeFrame);
+    const currentAmount = this.cache[key];
+    this.cache[key] = currentAmount ? currentAmount + amount : amount;
+  }
+
+  popAll() {
+    const toReturn = {};
+    Object.assign(toReturn, this.cache);
```

Comment:
Use object-assign ponyfill, import it like so:
`import objectAssign from 'object-assign';`